src/native/configuration/globalconfig.rs

```diff
@@ -74,15 +74,12 @@ impl GlobalConfig {
 
     // Get the Gredentials
     async fn get_credentials(&self) -> Result<Credentials, NativeError> {
-        if let Ok(signing_key) = self.api_key().await
-            && let Some(user_id) = self.remote_user_id
-        {
-            Ok(Credentials {
+        match (self.api_key().await, self.remote_user_id) {
+            (Ok(signing_key), Some(user_id)) => Ok(Credentials {
                 signing_key,
                 user_id,
-            })
-        } else {
-            Err(NativeError::missing_credentials())
+            }),
+            _ => Err(NativeError::missing_credentials()),
```

Comment:
The tuple construction should be delayed here and the two failure conditions should be teased apart into independent errors. Errors should represent a unique failure point that can easily be teased apart.